fog/overseer/server/tests/utils/mod.rs

```diff
@@ -0,0 +1,172 @@
+use mc_watcher::watcher_db::WatcherDB;
+use rand_core::{CryptoRng, RngCore};
+use std::{
+    collections::BTreeSet,
+    path::Path,
+    str::FromStr,
+    time::{Duration, SystemTime},
+};
+use tempdir::TempDir;
+use url::Url;
+
+const OMAP_CAPACITY: u64 = 4096;
+const BASE_PORT: u16 = 4997;
+
+// Helper which makes URIs and responder id for i'th server
+pub fn make_uris(idx: u16) -> (ResponderId, FogIngestUri, IngestPeerUri) {
+    let base_port = BASE_PORT + 10 * idx;
+
+    let local_node_id = ResponderId::from_str(&format!("0.0.0.0:{}", base_port + 5)).unwrap();
+    let client_listen_uri =
+        FogIngestUri::from_str(&format!("insecure-fog-ingest://0.0.0.0:{}/", base_port + 4))
+            .unwrap();
+    let peer_listen_uri =
+        IngestPeerUri::from_str(&format!("insecure-igp://0.0.0.0:{}/", base_port + 5)).unwrap();
+
+    (local_node_id, client_listen_uri, peer_listen_uri)
+}
+
+// Helper which makes i'th server and temp dir for its stuff (deleted when
+// objects are dropped)
+pub fn make_node(
+    idx: u16,
+    peer_idxs: impl Iterator<Item = u16>,
+    db: SqlRecoveryDb,
+    watcher_path: &Path,
+    ledger_db_path: &Path,
+    logger: Logger,
+) -> (
+    IngestServer<AttestClient, SqlRecoveryDb>,
+    TempDir,
+    FogIngestUri,
+) {
+    let logger = logger.new(o!("mc.node_id" => idx.to_string()));
+
+    let (local_node_id, client_listen_uri, peer_listen_uri) = make_uris(idx);
+
+    let peers: BTreeSet<IngestPeerUri> = peer_idxs.map(|idx| make_uris(idx).2).collect();
+
+    let state_tmp = TempDir::new("ingest_state").expect("Could not make tempdir for ingest state");
+    let state_file = state_tmp.path().join(format!("mc-ingest-state-{}", idx));
+
+    let config = IngestServerConfig {
+        ias_spid: Default::default(),
+        local_node_id,
+        peer_listen_uri,
+        peers,
+        fog_report_id: Default::default(),
+        client_listen_uri: client_listen_uri.clone(),
+        max_transactions: 10_000,
+        pubkey_expiry_window: 10,
+        peer_checkup_period: Some(Duration::from_secs(5)),
+        watcher_timeout: Duration::from_secs(5),
+        state_file: Some(StateFile::new(state_file)),
+        enclave_path: get_enclave_path(mc_fog_ingest_enclave::ENCLAVE_FILE),
+        omap_capacity: OMAP_CAPACITY,
+    };
+
+    // Open the Watcher DB
+    let watcher = WatcherDB::open_ro(watcher_path, logger.clone()).unwrap();
+
+    // Open the ledger db
+    let ledger_db = LedgerDB::open(ledger_db_path).unwrap();
+
+    let ra_client = AttestClient::new("").expect("Could not create IAS client");
+    let mut node = IngestServer::new(config, ra_client, db, watcher, ledger_db, logger);
+    node.start().expect("couldn't start node");
+
+    (node, state_tmp, client_listen_uri)
+}
+
+// Add an arbitrary block to ledger and a timestamp for it
+// This method is used in integration tests, but cargo still complains that it's
+// not in use.
+#[allow(dead_code)]
```

Comment:
Is this needed?

Reply:
Yes, CI complains that it's not used (I don't think cargo checks that it's being used in the tests/ directory).